From 7399eb9857a732987c34d09add2d461324a64754 Mon Sep 17 00:00:00 2001
From: Stephano Cetola <stephano@cetola.net>
Date: Thu, 9 Jul 2026 18:15:30 +0000
Subject: [PATCH 12/14] defaultenv: mnt-reform2-rk3588: fix disp-reset
 two-phase SPI read

The reform2_lpc protocol requires two separate SPI transactions with a
50ms gap. The write was already split, but the read half was missing.
Add the spi -r 8 call after the msleep to actually clock out the
RP2040 response; without it the read transaction never happens.

Signed-off-by: Stephano Cetola <stephanoc@gmail.com>
---
 .../defaultenv-mnt-reform2-rk3588/init/disp-reset               | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boards/mnt-reform2-rk3588/defaultenv-mnt-reform2-rk3588/init/disp-reset b/arch/arm/boards/mnt-reform2-rk3588/defaultenv-mnt-reform2-rk3588/init/disp-reset
index e7b1fc79d7..314cb76f18 100644
--- a/arch/arm/boards/mnt-reform2-rk3588/defaultenv-mnt-reform2-rk3588/init/disp-reset
+++ b/arch/arm/boards/mnt-reform2-rk3588/defaultenv-mnt-reform2-rk3588/init/disp-reset
@@ -17,3 +17,5 @@
 # Response: 8 bytes (ignored); RESX pulse occurs after the response is sent.
 spi -b 0 -c 0 -m 1 -f 1000000 0xb5 0x72 0x01 0xc6
 msleep 50
+spi -b 0 -c 0 -m 1 -f 1000000 -r 8
+msleep 50
-- 
2.43.0

